File: app/src/main/java/com/sumian/hw/widget/ReportSleepDurationView.java
package com.sumian.hw.widget;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import com.sumian.sd.R;
import com.sumian.hw.common.util.TimeUtil;
import com.sumian.hw.widget.base.BaseBlueCardView;

/**
 * Created by sm
 * on 2018/3/8.
 * desc:
 */

public class ReportSleepDurationView extends BaseBlueCardView {

    TextView mTvSleepTodayDuration;
    SleepStateDurationView mLightSleepDurationView;
    SleepStateDurationView mDeepSleepDurationView;

    public ReportSleepDurationView(@NonNull Context context) {
        this(context, null);
    }

    public ReportSleepDurationView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public ReportSleepDurationView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected int getLayoutRes() {
        return R.layout.hw_lay_sleep_duration_view;
    }

    @Override
    protected void initView(View inflate) {
        super.initView(inflate);

        mTvSleepTodayDuration = inflate.findViewById(R.id.tv_sleep_today_duration);
        mLightSleepDurationView = inflate.findViewById(R.id.light_sleep_duration_view);
        mDeepSleepDurationView = inflate.findViewById(R.id.deep_sleep_duration_view);
    }

    public void setSleepTodayDuration(int sleepTodayDuration) {
        CharSequence charSequence = TimeUtil.formatSleepDurationText(getContext(), sleepTodayDuration);
        mTvSleepTodayDuration.setText(charSequence);
    }

    public void setDeepSleepData(int deepSleepDuration, int deepSleepDurationPercent) {
        mDeepSleepDurationView.setData("深睡时长", deepSleepDuration, deepSleepDurationPercent);
    }

    public void setLightSleepData(int lightSleepDuration, int lightSleepDurationPercent) {
        mLightSleepDurationView.setData("浅睡时长", lightSleepDuration, lightSleepDurationPercent);
    }

}

File: app/src/main/java/com/sumian/hw/report/widget/ReportSleepDurationView.kt
package com.sumian.hw.report.widget

import android.content.Context
import android.util.AttributeSet
import com.sumian.hw.widget.base.BaseBlueCardView
import com.sumian.sd.R
import kotlinx.android.synthetic.main.hw_lay_sleep_duration_view.view.*

/**
 * Created by sm
 * on 2018/3/8.
 * desc:
 */

class ReportSleepDurationView : BaseBlueCardView {

    constructor(context: Context) : this(context, null)
    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)


    override fun getLayoutRes(): Int {
        return R.layout.hw_lay_sleep_duration_view
    }

    fun setSleepTodayDuration(sleepTodayDuration: Int) {
        tv_sleep_today_duration.setDuration(sleepTodayDuration)
    }

    fun setDeepSleepData(deepSleepDuration: Int, deepSleepDurationPercent: Int) {
        deep_sleep_duration_view.setData(resources.getString(R.string.deep_sleep_duration_hint), deepSleepDuration, deepSleepDurationPercent)
    }

    fun setLightSleepData(lightSleepDuration: Int, lightSleepDurationPercent: Int) {
        light_sleep_duration_view.setData(resources.getString(R.string.light_sleep_duration_hint), lightSleepDuration, lightSleepDurationPercent)
    }

}
